rad/src/types/mod.rs

```diff
@@ -466,13 +466,25 @@ impl TryFrom<CborValue> for RadonTypes {
 pub fn serial_iter_decode<T>(
     iter: &mut dyn Iterator<Item = (&[u8], &T)>,
     err_action: fn(RadError, &[u8], &T) -> Option<RadonReport<RadonTypes>>,
+    active_wips: &ActiveWips,
 ) -> Vec<RadonReport<RadonTypes>> {
     iter.filter_map(
         |(slice, inner)| match panic::catch_unwind(|| RadonTypes::try_from(slice)) {
-            Ok(Ok(radon_types)) => Some(RadonReport::from_result(
-                Ok(radon_types),
-                &ReportContext::default(),
-            )),
+            Ok(Ok(radon_types)) => {
+                // Handle future errors that should not appear yet
+                if let RadonTypes::RadonError(radon_error) = &radon_types {
+                    if let RadError::EncodeReveal = radon_error.inner() {
+                        if active_wips.wip0026() {
```

Comment:
This is backwards, we want to decode a MalformedReveal before WIP activation, not after.

```suggestion
                        if !active_wips.wip0026() {
```